### ISSUE TITLE
Support value conversions from two properties into one column

### ISSUE THREAD

**Issue by bcallaghan-et** [dotnet/efcore#20582]
I have a several classes that contain property pairs like the following
```cs
class MyEntity
{
    public int MyValue { get; set; }
    public bool MyValueSpecified { get; set; }
}
```
I would like EF to ignore `MyValue` and store `null` in its place if `MyValueSpecified` is false. Otherwise, `MyValue` should be mapped to the database as normal. The column behind this pairing should be nullable, but otherwise match the type of `MyValue`. To do this, I propose a new method on the model builder that allows multiple properties mapped to the same column. For example:
```cs
modelBuilder.Entity<MyEntity>()
    .Property(entity => new { entity.MyValue, entity.MyValueSpecified })
    .HasValueConversion(
        x => x.MyValueSpecified ? x.MyValue : null,
        x => x is null ? (false, default) : (true, x)
    );
```

I realize that this looks a lot like nullable value types, and I would rather use `int?` in my entities, but these classes are auto-generated by WCF and are outside of my control.

Implementing this feature probably depends on #13850.

**Comment by ajcvickers** [comment]
Similar to #13947. They should probably both be implemented together,

**Comment by AndriySvyryd** [comment]
Duplicate of #20582
